cartography/intel/gcp/firestore.py

```diff
@@ -0,0 +1,236 @@
+        request = firestore.projects().databases().list(parent=f"projects/{project_id}")
+        while request is not None:
+            response = request.execute()
+            if response.get('databases', []):
+                for database in response['databases']:
+                    database['id'] = database['name']
+                    firestore_databases.append(database)
+        return firestore_databases
+    except HttpError as e:
+        err = json.loads(e.content.decode('utf-8'))['error']
+        if err.get('status', '') == 'PERMISSION_DENIED' or err.get('message', '') == 'Forbidden':
+            logger.warning(
+                (
+                    "Could not retrieve Firestore Databases on project %s due to permissions issues. Code: %s, Message: %s"
+                ), project_id, err['code'], err['message'],
+            )
+            return []
+        else:
+            raise
+
+
+@timeit
+def get_firestore_indexes(firestore: Resource, firestore_databases: List[Dict], project_id: str) -> List[Dict]:
+    """
+        Returns a list of firestore indexes for a given project.
+
+        :type firestore: Resource
+        :param firestore: The firestore resource created by googleapiclient.discovery.build()
+
+        :type firestore_databases: list
+        :param firestore_database: A list of firestore databases
+
+        :type project_id: str
+        :param project_id: Current Google Project Id
+
+        :rtype: list
+        :return: List of Firestore Indexes
+    """
+    for database in firestore_databases:
+        try:
+            firestore_indexes = []
+            request = firestore.projects().databases().collectionGroups().indexes().list(parent=f"{database['name']}/collectionGroups/*")
+            while request is not None:
+                response = request.execute()
+                if response.get('indexes', []):
+                    for index in response['indexes']:
+                        index['database_id'] = database['id']
+                        index['id'] = index['name']
+                        firestore_indexes.append(index)
+                request = firestore.projects().databases().collectionGroups().indexes().list_next(previous_request=request, previous_response=response)
+            return firestore_indexes
+        except HttpError as e:
+            err = json.loads(e.content.decode('utf-8'))['error']
+            if err.get('status', '') == 'PERMISSION_DENIED' or err.get('message', '') == 'Forbidden':
+                logger.warning(
+                    (
+                        "Could not retrieve Firestore Indexes on project %s due to permissions issues. Code: %s, Message: %s"
+                    ), project_id, err['code'], err['message'],
+                )
+                return []
+            else:
+                raise
+
+
+@timeit
+def load_firestore_databases(session: neo4j.Session, data_list: List[Dict[str, Optional[str]]], project_id: str, update_tag: int) -> None:
```

Comment:
change the type for data_list. similarly for other methods. the type should be `List[Dict]`